libs/datamodel/core/src/ast/parser/parse_directive.rs

```diff
@@ -1,16 +1,22 @@
-use super::{helpers::ToIdentifier, parse_expression::parse_arg_value, Rule};
+use super::{
+    helpers::{parsing_catch_all, ToIdentifier, Token, TokenExtensions},
+    parse_expression::parse_arg_value,
+    Rule,
+};
 use crate::ast::*;
 
-pub fn parse_directive(token: &pest::iterators::Pair<'_, Rule>) -> Directive {
+pub fn parse_directive(token: &Token) -> Directive {
     let mut name: Option<Identifier> = None;
     let mut arguments: Vec<Argument> = vec![];
 
-    match_children! { token, current,
-        Rule::directive => return parse_directive(&current),
-        Rule::directive_name => name = Some(current.to_id()),
-        Rule::directive_arguments => parse_directive_args(&current, &mut arguments),
-        _ => unreachable!("Encountered impossible directive during parsing: {:?} \n {:?}", token, current.tokens())
-    };
+    for current in token.filtered_children().into_iter() {
```

Comment:
I think the `into_iter()` is redundant here and in other for loops in this PR (the `for` desugaring already calls that)

Reply:
ah yeah you are right. Thanks 🙏